### PR TITLE
Portfolio

### PR DESCRIPTION
**Pull request by Nomdecyber**
Addresses issue #8.

### CODE REVIEW

File: stockticker/templates/index.html [hunk 1]
```diff
@@ -19,24 +19,29 @@
     <link
         href="https://cdn.pydata.org/bokeh/release/bokeh-tables-0.12.{{ bokeh }}.min.css"
         rel="stylesheet" type="text/css">
-
+    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
```
Comment (ZachGlassman) [inline stockticker/templates/index.html]: We should only have a single verson of jquery loaded, we have a diffferent version loaded below

File: stockticker/stockticker.py [hunk 2]
```diff
@@ -32,16 +33,75 @@ def query_quandl(ticker, quandl_key, value='open', days=500):
...
+    """Makes a query to Prophet"""
+    prophet_params = {'ds': dates, 'y': prices, 'key': secret_key}
+
+    return requests.post(prophet_url, data=json.dumps(prophet_params))
+
+def get_error_div(error_string):
+    """Returns a div displaying an error message"""
+    return "<div>" + error_string + "</div>"
+
+def get_initial_js(ticker_handler_url, portfolio_handler_url,
```
Comment (ZachGlassman) [inline stockticker/stockticker.py]: this function can be done within flask templating, I think it will be somewhat cleaner to do it that way.

**Comment by ZachGlassman** [comment]
So this is how this works now

1. Front end says "get these tickers"
2. Backend sends template with javascript code to frontend
3. frontend queries each ticker route (asynchronously)
4. front end aggregates results and when all gathered queries backend again for aggregate
5. backend returns plot

I think we are better off eliminating steps 2-4. Lets use the interface that the user sees to eliminate steps 1 and 2.  When a user presses click on a page to see individual stocks, we can have each stock entry element query its own route using JS (pretty much the code you already have).  On a portfolio page, they can query a portfolio route directly which will get rid of step 4.  

This way for the case of multiple views, we don't need to have as many passes between front end and back end.  For the portfolio case, we don't really care that the information is updated as it arrives because the end result is what we care about and I would prefer to do that all on the backend as opposed to passing the results to the frontend and then back to the backend and back.



We should be able to reuse most of the JS code you have already written except it will be templated into the frontend, however, if you want a cleaner way of handling frontend state, you might want to check out the React library.

File: stockticker/templates/index.html (hunk 1 above)
Comment (Nomdecyber) [reply stockticker/templates/index.html]: Not sure how I missed that. Fixed in commit 8e18ad77e03d332b7434dbe7da50f45cac14b602

File: stockticker/stockticker.py (hunk 2 above)
Comment (Nomdecyber) [reply stockticker/stockticker.py]: Agreed. Fixed in commit 8e18ad77e03d332b7434dbe7da50f45cac14b602.

**Comment by ZachGlassman** [comment]
This is looking good!  Lets fix the failing tests. and then we should be good to go.  The test that is failing is not a useful one, it was mostly there as an illustration, so I'm fine with it being taken out or modified to be actually useful.

**Comment by ZachGlassman** [comment]
Since we don't have a staging environment (although we could set that up), let's merge this in after we rebase.

**Pull request merged**